src/types.mligo

```diff
@@ -122,14 +122,17 @@ type proposal =
 type seconds = nat
 type voting_period = { length : seconds }
 
+// For efficiency, we only keep a `nat` for the numerator, whereas the
+// denominator is not stored and has a fixed value of `1000000`.
+type quorum_fraction = { numerator : int }
+let quorum_denominator = 1000000
+let percent_to_quorum = 10000
```

Comment:
@sras [here](https://github.com/tqtezos/baseDAO/pull/225#discussion_r632692172) you said:
> Have changed quorum threshold to be an Int.

but looks like you changed `quorum_threshold` to be a `nat` and `quorum_fraction` to have an `int` in it?

I don't get the reason for this change, my point was that in these types we be better off using a `nat` as we want to
verify that the value isn't negative.

What I meant was that we can use `int`s as an intermediate type in the calculations of `update_quorum` without wrapping them in a `numerator` to be a `quorum_threshold` or `quorum_fraction` (as we indeed do now).

But IMO here we can have:
```suggestion
// For efficiency, we only keep a `nat` for the numerator, whereas the
// denominator is not stored and has a fixed value of `1000000`.
type quorum_fraction = { numerator : nat }
let quorum_denominator = 1000000n
let percent_to_quorum = 10000n
```
and just below do:
```ocaml
type quorum_threshold = quorum_fraction
```

and AFAICT `update_quorum` would remain the same, aside from moving some calculation of `int()`s.

Does this make sense to you?

Reply:
@pasqu4le It generally makes sense to me. But I don't really think it is a good idea to use plain `int` as intermediate for calculations (eventhough that is how it has been implemented in the last commit I pushed yesterday). AFAIU we can use a record with a single field for this without incurring no additional cost. The reasoning for this is that it is hard to track `int`s that should be treated as a numerator of a fraction, and ones that should be treated as such. I think a comment is too ineffective to consistently provide this distinction in a code base that is still evolving.

So what I have now is two types, `unsigned_quorum_fraction` which uses a `nat` for numerator and `quorum_fraction` which uses an `int` for the same. We use `unsigned_quorum_fraction` every where in storage and config, and use `quorum_fraction` for doing computation. As I said earlier, AFAIU it should be completely free, and the same as using raw `int`s, so long as we inline functons like `fsub` and `fadd`.

Reply:
That's a great point, using two different "newtype"s seems like the most elegant solution to me :ok_hand: